Library/Homebrew/cmd/install.rb

```diff
@@ -341,4 +342,28 @@ def install_formula(f)
   rescue CannotInstallFormulaError => e
     ofail e.message
   end
+
+  def env_forbidden_licenses
+    Homebrew::EnvConfig.forbidden_licenses.split(" ")
+  end
+
+  def forbidden_license_check(f)
+    forbidden_licenses = env_forbidden_licenses
+
+    if forbidden_licenses.include? f.license
+      raise CannotInstallFormulaError, <<~EOS
+        #{f.name} has a forbidden license #{f.license}.
+      EOS
+    end
+
+    fi = FormulaInstaller.new(f)
+    fi.compute_dependencies.each do |dep, _|
+      dep_f = dep.to_formula
+      next unless forbidden_licenses.include? dep_f.license
+
+      raise CannotInstallFormulaError, <<~EOS
+        The installation of #{f.name} has a dependency on #{dep.name} with a forbidden license #{dep_f.license}.
+      EOS
+    end
+  end
```

Comment:
I'd consider moving all of this to `formula_installer.rb` given it's having to make use of it anyway.